### DISCUSSION TITLE
Why do all option builder structs take Buffer or Window arguments by moving?

### DISCUSSION THREAD

**Discussion by dasupradyumna** [noib3/nvim-oxi#199] [General]
For example, if I create a buffer /  window and want to set options, create autocommands and commands, then when I use the corresponding option builder structs for building the option structs, I noticed that `Buffer` and `Window` are moved into the `buffer()` and `win()` methods. So I have to use `clone()` every time I want to use these.

What is the motivation behind moving them and not passing them as a reference?  
I believe passing them as references would simplify the user API. Are there any demerits to this?

**Comment by noib3** [comment]
The bodies of the functions you pass `Buffer`s and `Window`s into operate on owned instances, so it would be suboptimal for them to take e.g. a `&Buffer` only to immediately `clone()` it.

There were talks of making `Buffer`, `Window` and `TabPage` `Copy` types, which would alleviate the syntax burden.

**Reply by dasupradyumna** [reply]
I see, that makes sense. Was the conclusion of those talks to proceed with making them `Copy` types?
Are there any plans towards implementing this?
